ktor-client/ktor-client-plugins/ktor-client-logging/common/src/io/ktor/client/plugins/logging/Logging.kt

```diff
@@ -69,15 +89,342 @@ public class LoggingConfig {
+
+            body is OutgoingContent.WriteChannelContent ||
+                body is OutgoingContent.ReadChannelContent -> "--> ${request.method.value} $uri (unknown-byte body)"
+
+            else -> {
+                val size = computeRequestBodySize(request.body, headers)
+                "--> ${request.method.value} $uri ($size-byte body)"
+            }
+        }
+
+        logger.log(startLine)
+
+        if (!isHeaders() && !isBody()) {
+            return null
+        }
+
+        for ((name, values) in headers.entries()) {
+            if (sanitizedHeaders.find { sh -> sh.predicate(name) } == null) {
+                logger.log("$name: ${values.joinToString(separator = ", ")}")
+            } else {
+                logger.log("$name: ██")
+            }
+        }
+
+        if (!isBody() || request.method == HttpMethod.Get || request.method == HttpMethod.Head) {
+            logger.log("--> END ${request.method.value}")
+            return null
+        }
+
+        logger.log("")
+
+        if (body !is OutgoingContent) {
+            logger.log("--> END ${request.method.value}")
+            return null
+        }
+
+        val newContent = if (request.headers[HttpHeaders.ContentEncoding] == "gzip") {
+            logOutgoingContent(body, request.method, headers) { channel ->
+                GZipEncoder.decode(channel)
+            }
+        } else {
+            logOutgoingContent(body, request.method, headers)
+        }
+
+        return newContent
+    }
+
+    suspend fun logResponseBody(response: HttpResponse, body: ByteReadChannel) {
+        logger.log("")
+
+        val (isBinary, size, newBody) = detectIfBinary(
+            body,
+            response.contentLength(),
+            response.contentType(),
+            response.headers
+        )
+        val duration = response.responseTime.timestamp - response.requestTime.timestamp
+
+        if (size == 0L) {
+            logger.log("<-- END HTTP (${duration}ms, $size-byte body)")
+            return
+        }
+
+        if (!isBinary) {
+            val contentType = response.contentType()
+            val charset = if (contentType != null) {
+                contentType.charset() ?: Charsets.UTF_8
+            } else {
+                Charsets.UTF_8
+            }
+
+            logger.log(newBody.readRemaining().readText(charset = charset))
+            logger.log("<-- END HTTP (${duration}ms, $size-byte body)")
+        } else {
+            var type = "binary"
+            if (response.headers.contains(HttpHeaders.ContentEncoding)) {
+                type = "encoded"
+            }
+
+            if (size != null) {
+                logger.log("<-- END HTTP (${duration}ms, $type $size-byte body omitted)")
+            } else {
+                logger.log("<-- END HTTP (${duration}ms, $type body omitted)")
+            }
+        }
+    }
+
+    suspend fun logResponseStdFormat(response: HttpResponse): HttpResponse {
```

Comment:
Probably, should be renamed
```suggestion
    suspend fun logResponseOkHttpFormat(response: HttpResponse): HttpResponse {
```